pkg/api/object_example_test.go

```diff
@@ -42,13 +43,36 @@ func (o *ExampleObject) EndpointURL(ctx context.Context) (*url.URL, error) {
 	return url.Parse("/example/v1")
 }
 
+// GetIdentifier returns the identifier of an object.
+// The Makefiles `generate` target creates this method for objects that have a anxcloud:"identifier" tagged field.
+// Objects with multiple identifying fields can have this method overwritten to return an operation dependent identifier.
+// The provided context might not always have the operation set. In this case the method must return the primary identifier.
+//
+// For example: `corev1.Location` has the (primary) `Identifier` attribute, which is required by many API endpoints.
+// Additionally locations are identified by the human readable `Code` attribute (e.g. ANX04).
+//
+// Handling of secondary identifiers usually requires additional logic in the objects EndpointURL method.
+func (o *ExampleObject) GetIdentifier(ctx context.Context) (string, error) {
+	op, err := types.OperationFromContext(ctx)
+	if err == nil && op == types.OperationGet {
+		return o.SecondaryIdentifier, nil
+	}
```

Comment:
shouldn't this also check if `o.Identifier` is set and if so, return that and only otherwise look at secondary identifiers?

Reply:
you made this really nice for corev1.Location, checking if Identifier is set, if so returning it and only otherwise go to the secondary identifier logic. I'd love this example to look as great as that.

Also, please add the custom EndpointURL logic required for this to this example, too.